//= error assert
fun main() {
  var x = Foo(12);
  x.test();
  x.test2(1);
}

class Foo(let i: Int32) {
  fun test() {
    var x = self.i;
  }

  fun test2(i: Int32) {
    assert(i == 2);
  }
}